update_prevalence.py

```diff
@@ -911,99 +911,127 @@ def rolldown_vaccine_types(parent: Place, children: List[Place]):
+                            raise ValueError(f"Missing population data for {county!r}")
+                    rollup_population(state, "counties")
+                    rollup_vaccines(state, "counties")
+                rollup_population(country, "states")
+            except ValueError as err:
+                failed_countries.append(dict(country_name=country.name, error=err))
+
+        # Delete any countries that had an error
+        for country_data in failed_countries:
+            print_and_log_to_sentry(
+                f"Discarding country {country_data['country_name']} due to error: {country_data['error']!r}"
+            )
+            del self.countries[country_data["country_name"]]
+
+        # Clear out the list for the next pass below
+        failed_countries = list()
 
         for country in list(self.countries.values()):
-            for state in list(country.states.values()):
-                for county in list(state.counties.values()):
-                    if not county.cumulative_cases:
-                        if (
-                            county.fullname
-                            in (
-                                # These just don't have any reported cases
-                                "Hoonah-Angoon, Alaska, US",
-                                "Lake and Peninsula, Alaska, US",
-                                "Skagway, Alaska, US",
-                                "Unassigned, District of Columbia, US",
-                                "Kalawao, Hawaii, US",
-                                # These are reported under a combined name
-                                "Dukes, Massachusetts, US",
-                                "Nantucket, Massachusetts, US",
-                                "Bronx, New York, US",
-                                "Kings, New York, US",
-                                "Queens, New York, US",
-                                "Richmond, New York, US",
-                                # Utah reports by region, not county
-                            )
-                            or county.state == "Utah"
-                        ):
-                            pass  # don't warn
-                        else:
-                            print_and_log_to_sentry(f"Discarding {county!r} with no case data")
-                        del state.counties[county.name]
-
-                    if county.test_positivity_rate is None:
-                        if county.tests_in_past_week:
-                            county.test_positivity_rate = county.cases_last_week / county.tests_in_past_week
-                            if county.cases_last_week > county.tests_in_past_week:
-                                print_and_log_to_sentry(
-                                    f"Falling back to state data for {county.name} because it has more cases than tests in the last week with a positivity rate of {county.test_positivity_rate}."
+            try:
```

Comment:
These try-catch blocks suggest the contents could be more readable as a helper function.

Reply:
Good point! I have limited time this week, so perhaps we can merge this PR and I'll do this as a follow on?

Reply:
SGTM. 

Another option is to use the population from the last successful update, if applicable.